src/compiler/utilities.ts

```diff
@@ -2083,6 +2083,12 @@ namespace ts {
         return start <= textSpanEnd(span) && end >= span.start;
     }
 
+    export function decodedTextSpanIntersectsWith(start1: number, length1: number, start2: number, length2: number) {
+        let end1 = start1 + length1;
+        let end2 = start2 + length2;
+        return start2 <= end1 && end2 >= start1;
+    }

```

Comment:
"Overlaps" has very different semantics from "Intersects" from the perspective of the Editor and how they use that terminology.  This is following the editor semantics for 'intersects', so i'd like to keep using that name.